Submitted page

zineanteoh opened the pull request:
Front page
<img width="1432" alt="Screen Shot 2022-09-12 at 10 23 02 PM" src="https://user-images.githubusercontent.com/58854510/189800490-da5d0ede-cdf1-4e9e-a5a2-9d12cd9365e9.png">

After logging in for the first time as a hacker
<img width="1431" alt="Screen Shot 2022-09-12 at 10 23 26 PM" src="https://user-images.githubusercontent.com/58854510/189800521-90d36c03-2261-4ccb-ad8b-1dd5aa9657af.png">

After submitting the application form
<img width="1429" alt="Screen Shot 2022-09-12 at 10 22 39 PM" src="https://user-images.githubusercontent.com/58854510/189800445-45ed5ac4-1bec-4026-9b75-5f6d3519929c.png">

aadibajpai commented:
I think the "Not you" text is not necessary, removing that would allow us to inline the text and the button

zineanteoh commented:
@aadibajpai ready to be reviewed & merged. 

Fixed: 
- Added file size limit (1 MB)
- Added validation for uploaded pdf (previously when user remove uploaded file they could proceed to submit form)
- Repeating API calls (caused by useSWR revalidatingOnFocus)
- Some stylings (register, submit button, etc.)
- Some other form stuffs (date of birth > 18 years, etc.)

aadibajpai commented on pages/HackerDash.tsx:
```diff
@@ -1,27 +1,42 @@
+import moment from 'moment';
 
-export default function HackerDash() {
+type Props = {
```
probably should be more descriptive, HackerProps or sth

aadibajpai commented on pages/HackerDash.tsx:
```diff
@@ -1,27 +1,42 @@
+		{ revalidateOnFocus: false, revalidateOnMount: true }
+	);
+
+	if (user && user.applicationStatus) {
```
```suggestion
	if (user?.applicationStatus) {
```

aadibajpai commented on pages/HackerDash.tsx:
```diff
@@ -61,6 +85,36 @@ export default function HackerDash() {
+	const [resumeFile, setResumeFile] = useState<UploadFile[]>([]);
+	const [resumeToUpload, setResumeToUpload] = useState<string | Blob>('');
+
+	const dummyRequest = (options: any) => {
```
why do we need them? should add a comment

aadibajpai commented on pages/index.tsx:
```diff
@@ -1,30 +1,40 @@
+						<HackerDash
+							userApplicationStatus={userApplicationStatus}
+							setUserApplicationStatus={setUserApplicationStatus}
+							userEmail={session.user?.email}
```
at this juncture why would user not have email?

zineanteoh replied:
presumably because they haven't logged in yet (happy to change if there's a better approach)

aadibajpai commented:
> @aadibajpai ready to be reviewed & merged.
> 
> Fixed:
> 
> * Added file size limit (1 MB)
> * Added validation for uploaded pdf (previously when user remove uploaded file they could proceed to submit form)
> * Repeating API calls (caused by useSWR revalidatingOnFocus)
> * Some stylings (register, submit button, etc.)
> * Some other form stuffs (date of birth > 18 years, etc.)
> 
> <img alt="Screen Shot 2022-09-14 at 1 13 55 AM" width="1430" src="https://user-images.githubusercontent.com/58854510/190074011-104fe0a9-4305-4f28-8160-5464fce482e3.png">
> 
> <img alt="Screen Shot 2022-09-14 at 1 14 54 AM" width="1434" src="https://user-images.githubusercontent.com/58854510/190074192-7b56a569-71bb-46f0-a642-858f70b6ecf9.png">
> 
> <img alt="Screen Shot 2022-09-14 at 1 15 22 AM" width="1437" src="https://user-images.githubusercontent.com/58854510/190074216-2d84dd29-5370-4d05-bc9d-fd9151bdc212.png">

looks good I left some comments. We should probably bump limit to 10mb or so, to not miss any legit resumes which for some reason are > 1mb

zineanteoh commented:
Change to a more fitting logo

<img width="1432" alt="Screen Shot 2022-09-14 at 1 52 24 AM" src="https://user-images.githubusercontent.com/58854510/190082054-2cb68139-5eba-4b28-a52f-5c6c7308897e.png">

aadibajpai commented:
> Change to a more fitting logo
> 
> <img alt="Screen Shot 2022-09-14 at 1 52 24 AM" width="1432" src="https://user-images.githubusercontent.com/58854510/190082054-2cb68139-5eba-4b28-a52f-5c6c7308897e.png">

looks worse imo, we should revert to the previous one. It's not aligned well either
